src/app/trainee/revalidation-history/revalidation-history.component.html

```diff
@@ -51,17 +58,16 @@
             <mat-form-field class="w-100">
               <mat-label>Make a recommendation</mat-label>
               <mat-select formControlName="action">
-                <mat-option>None</mat-option>
-                <mat-option value="Revalidate">Revalidate</mat-option>
-                <mat-option value="Defer">Defer</mat-option>
-                <mat-option value="Non-Engagement">Non-Engagement</mat-option>
+                <mat-option *ngFor="let revalType of revalidationType | keyvalue" [value]="revalType.key">
+                  {{ revalType.value }}
+                </mat-option>
               </mat-select>
               <mat-error *ngIf="revalidationForm.controls['action']?.hasError('required')">
                 Please make a recommendation
               </mat-error>
             </mat-form-field>
 
-            <mat-form-field class="w-100" *ngIf="revalidationForm.controls['action']?.value === 'Defer'">
+            <mat-form-field class="w-100" *ngIf="revalidationForm.controls['action']?.value === 'DEFER'">
```

Comment:
can probs utilize the defer enum value here from a variable rather than a string in the template

Reply:
how? tried and it seems painful e.g. `revalidationType[revalidationForm.controls['action']?.value] === 'Defer'`
can you help with another way please?

Reply:
in the ts component you could create a variable that maps to the enum and then reference it in your template. i've done something similar on line 11 of `src/app/trainees/trainee-filters/trainee-filters.component.html`

Reply:
yup, that maps "underNotice" == FILTER_TYPE which is enum value "underNotice". I'm passing enum back to back-end and comparing drop-down value enumKey === enumKey.

Reply:
fixed, but must say it looks ugly :-)

Reply:
done

Reply:
thanks for addressing this as per our slack call